flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/Checkpoint.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.checkpoint;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.runtime.jobgraph.JobVertexID;
+import org.apache.flink.util.Preconditions;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * A successful checkpoint describes a checkpoint after all required tasks acknowledged it (with their state)
+ * and that is considered completed.
+ */
+public class Checkpoint implements Serializable {
+
+	private static final long serialVersionUID = -8360248179615702014L;
+
+	private final JobID job;
+	
+	private final long checkpointID;
+
+	/** The timestamp when the checkpoint was triggered. */
+	private final long timestamp;
+
+	/** The duration of the checkpoint (completion timestamp - trigger timestamp). */
+	private final long duration;
+
+	/** States of the different task groups belonging to this checkpoint */
+	private final Map<JobVertexID, TaskState> taskStates;
+
+	public Checkpoint(
+		JobID job,
+		long checkpointID,
+		long timestamp,
+		long completionTimestamp,
+		Map<JobVertexID, TaskState> taskStates) {
+
+		this.job = job;
```

Comment:
Will do (this is an unrelated change though as `Checkpoint` is a renaming of the current `master`'s `CompletedCheckpoint`)